tdrs-backend/gunicorn_start.sh

```diff
@@ -25,9 +25,6 @@ sleep 5
 # celery -A tdpservice.settings --broker=$REDIS_URI flower &
 celery -A tdpservice.settings beat -l info --scheduler django_celery_beat.schedulers:DatabaseScheduler &
 
-# rebuild and repopulate elastic search indexes
-python manage.py search_index --rebuild -f
```

Comment:
removed this as it was causing the following issue when deploying

```
2023-01-25T09:17:24.24-0800 [APP/PROC/WEB/0] ERR [2023-01-25 17:17:24,248: INFO/MainProcess] DatabaseScheduler: Schedule changed.
   2023-01-25T09:17:24.31-0800 [APP/PROC/WEB/0] OUT Creating index 't1_submissions'
   2023-01-25T09:17:24.41-0800 [APP/PROC/WEB/0] OUT Creating index 't2_submissions'
   2023-01-25T09:17:24.51-0800 [APP/PROC/WEB/0] OUT Creating index 't4_submissions'
   2023-01-25T09:17:24.61-0800 [APP/PROC/WEB/0] OUT Creating index 't5_submissions'
   2023-01-25T09:17:24.73-0800 [APP/PROC/WEB/0] OUT Creating index 't6_submissions'
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] OUT Indexing 3288 'T1' objects
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR Traceback (most recent call last):
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR return self.cursor.execute(sql, params)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR psycopg2.errors.UndefinedColumn: column search_indexes_t1.record does not exist
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR LINE 1: ...OR WITH HOLD FOR SELECT "search_indexes_t1"."id", "search_in...
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR ^
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR The above exception was the direct cause of the following exception:
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR Traceback (most recent call last):
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR cursor.execute(sql, params)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 66, in execute
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR return executor(sql, params, many, context)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 79, in _execute
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR with self.db.wrap_database_errors:
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/utils.py", line 90, in __exit__
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR raise dj_exc_value.with_traceback(traceback) from exc_value
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/backends/utils.py", line 84, in _execute
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR return self.cursor.execute(sql, params)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR django.db.utils.ProgrammingError: column search_indexes_t1.record does not exist
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR LINE 1: ...OR WITH HOLD FOR SELECT "search_indexes_t1"."id", "search_in...
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR ^
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR During handling of the above exception, another exception occurred:
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR Traceback (most recent call last):
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/app/manage.py", line 31, in <module>
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR main()
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/app/manage.py", line 27, in main
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR execute_from_command_line(sys.argv)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR utility.execute()
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR self.fetch_command(subcommand).run_from_argv(self.argv)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR self.execute(*args, **cmd_options)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR output = self.handle(*args, **options)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django_elasticsearch_dsl/management/commands/search_index.py", line 159, in handle
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR self._rebuild(models, options)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django_elasticsearch_dsl/management/commands/search_index.py", line 140, in _rebuild
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR self._populate(models, options)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django_elasticsearch_dsl/management/commands/search_index.py", line 117, in _populate
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR doc().update(qs, parallel=parallel)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django_elasticsearch_dsl/documents.py", line 195, in update
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR return self._bulk(
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django_elasticsearch_dsl/documents.py", line 179, in _bulk
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR return self.bulk(*args, **kwargs)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django_elasticsearch_dsl/documents.py", line 146, in bulk
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR return bulk(client=self._get_connection(), actions=actions, **kwargs)
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/elasticsearch/helpers/actions.py", line 410, in bulk
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR for ok, item in streaming_bulk(
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/elasticsearch/helpers/actions.py", line 319, in streaming_bulk
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR for bulk_data, bulk_actions in _chunk_actions(
   2023-01-25T09:17:24.91-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/elasticsearch/helpers/actions.py", line 155, in _chunk_actions
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR for action, data in actions:
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django_elasticsearch_dsl/documents.py", line 170, in _get_actions
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR for object_instance in object_list:
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/models/query.py", line 353, in _iterator
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR yield from self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/models/query.py", line 51, in __iter__
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR File "/home/vcap/deps/1/python/lib/python3.10/site-packages/django/db/models/sql/compiler.py", line 1178, in execute_sql
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR cursor.close()
   2023-01-25T09:17:24.92-0800 [APP/PROC/WEB/0] ERR psycopg2.errors.InvalidCursorName: cursor "_django_curs_140039043520320_sync_1" does not exist
   2023-01-25T09:17:29.09-0800 [APP/PROC/WEB/0] ERR [2023-01-25 17:17:29,095: INFO/MainProcess] DatabaseScheduler: Schedule changed.
   2023-01-25T09:17:35.78-0800 [APP/PROC/WEB/0] ERR [2023-01-25 17:17:35 +0000] [726] [DEBUG] Closing connection.
   2023-01-25T09:18:01.11-0800 [HEALTH/0] ERR Timed out after 1m0s (30 attempts) waiting for readiness check to succeed: failed to make TCP connection to 10.255.59.197:8080: dial tcp 10.255.59.197:8080: connect: connection refused
   2023-01-25T09:18:01.11-0800 [CELL/0] ERR Failed after 1m0.388s: readiness health check never passed.
   2023-01-25T09:18:01.11-0800 [CELL/SSHD/0] OUT Exit status 0
   2023-01-25T09:18:05.81-0800 [APP/PROC/WEB/0] ERR [2023-01-25 17:18:05 +0000] [726] [DEBUG] Closing connection.
```

I don't believe this line is needed: `django-elasticsearch-dsl` will manage its own migrations and re-indexing anyway, there's no point in forcing it after deployments. I don't fully understand the root cause, but I believe this was a fact of calling the reindex function at the wrong time when database migrations hadn't fully taken effect.

Reply:
Sorry about that. I thought we had removed that line already.